### PR TITLE
support metric=rsa_* in get_linregress_consistency

### PR DESCRIPTION
**Pull request by Milotrince**

### CODE REVIEW

File: brainmodel_utils/metrics/consistency.py [hunk 1]
```diff
@@ -225,10 +238,11 @@ def get_linregress_consistency_persphalftrial(
 def get_linregress_consistency(
     source,
     target,
-    map_kwargs,
+    map_kwargs={ "map_type": "identity" },
```
Comment (anayebi) [inline brainmodel_utils/metrics/consistency.py]: Is it possible to not have a default mapping function? Especially since the identity mapping is a passthrough and not a "real" mapping function

File: brainmodel_utils/metrics/consistency.py [hunk 2]
```diff
@@ -229,25 +242,27 @@ def get_linregress_consistency(
     num_bootstrap_iters=1000,
     num_parallel_jobs=1,
```
Comment (anayebi) [inline brainmodel_utils/metrics/consistency.py:243]: num_parallel_jobs=1 should probably not be the default?
We should revert to the original default

File: brainmodel_utils/metrics/consistency.py [hunk 3]
```diff
@@ -229,25 +242,27 @@ def get_linregress_consistency(
...
-    Optional Arguments:
-    -------------------
-    metric: Correlation metric (across stimuli) used per neuron. Default is Pearson's R.
-            Other choices can be "spearmanr", for example.
-    splits: Your own list of {"train", "test"} split indices.
+    metric (optional): Correlation metric (across stimuli) used per neuron.
+            Supported metrics: "pearsonr", "spearmanr", "rsa_pearsonr", "rsa_spearmanr"
+    splits (optional): Your own list of {"train", "test"} split indices.
             If "splits" is None, you can additionally specify train_frac and num_train_test_splits to generate your own.
+            "rsa_*" metrics do not use train/test splits, so this is ignored in that case.
```
Comment (anayebi) [inline brainmodel_utils/metrics/consistency.py]: This should be updated to have that RSA does use train/test splits

File: brainmodel_utils/metrics/consistency.py [hunk 4]
```diff
@@ -287,5 +314,5 @@ def get_linregress_consistency(
             xarray_dims=["trial_bootstrap_iters", "train_test_splits", "units"],
         )
     else:
-        results_dict = concat_dict_sp(results_arr)
+        results_dict = concat_dict_sp(results_arr, partition_names=results_arr[0].keys())
```
Comment (anayebi) [inline brainmodel_utils/metrics/consistency.py]: Interesting, what does partition_names do? What error is thrown by not having it, as was the case before?

File: brainmodel_utils/metrics/consistency.py (hunk 2 above)
Comment (Milotrince) [reply brainmodel_utils/metrics/consistency.py:243]: This was the default

File: brainmodel_utils/metrics/consistency.py (hunk 4 above)
Comment (Milotrince) [reply brainmodel_utils/metrics/consistency.py]: concat_dict_sp expects partition=[train, test], so in the case that there is no train/test (which is no longer the case) this would throw an error

Comment (anayebi) [reply brainmodel_utils/metrics/consistency.py]: Gotcha, I guess it's now reverted back to `results_dict = concat_dict_sp(results_arr)` :)

File: brainmodel_utils/metrics/consistency.py (hunk 2 above)
Comment (anayebi) [reply brainmodel_utils/metrics/consistency.py:243]: Ok this looks good to me. Since you use this code more than me, do you personally think `num_parallel_jobs=1` should be set to something else? The upside is then the default isn’t something slow. The downside is that if the user doesn’t have at least `num_parallel_jobs` CPU cores, then things may also be slow lol. Just curious if it is cumbersome for you to have to remember to set it each time.

Comment (Milotrince) [reply brainmodel_utils/metrics/consistency.py:243]: setting the parallel jobs is not an issue, I use argparse to set a flag for all my scripts (e.g. --cpus=64)

**Review by anayebi — APPROVED** [review]
Looks good to me!
